File: server/backend/src/backend/api/models/media_download_history.py
from __future__ import annotations

from datetime import datetime
from typing import Any

from backend.api.models.base import ResponseBase


class MediaDownloadHistoryEntryRead(ResponseBase):
    id: int
    media_download_id: int
    action: str
    label: str
    status: str
    occurred_at: datetime
    duration_ms: int | None = None
    duration: str | None = None
    detail: str | None = None
    metadata: dict[str, Any]


class MediaDownloadHistoryPageRead(ResponseBase):
    items: list[MediaDownloadHistoryEntryRead]
    total: int
    offset: int
    limit: int
    has_more: bool

File: ui/src/types/schemas/media_download_history.ts
import {z} from 'zod'
import {ApiDateTimeStringSchema} from './datetime'

export const MediaDownloadHistoryEntryReadSchema = z.looseObject({
  id: z.int(),
  mediaDownloadId: z.int(),
  action: z.string(),
  label: z.string(),
  status: z.string(),
  occurredAt: ApiDateTimeStringSchema,
  durationMs: z.int().nullable(),
  duration: z.string().nullable(),
  detail: z.string().nullable(),
  metadata: z.record(z.string(), z.unknown()),
})

export const MediaDownloadHistoryPageReadSchema = z.looseObject({
  items: z.array(MediaDownloadHistoryEntryReadSchema),
  total: z.int(),
  offset: z.int(),
  limit: z.int(),
  hasMore: z.boolean(),
})

export type MediaDownloadHistoryEntryRead = z.infer<typeof MediaDownloadHistoryEntryReadSchema>
export type MediaDownloadHistoryPageRead = z.infer<typeof MediaDownloadHistoryPageReadSchema>
